internal/driver/nodeserver.go

```diff
@@ -54,27 +52,22 @@ func (ns *LinodeNodeServer) NodePublishVolume(ctx context.Context, req *csi.Node
 	defer ns.mux.Unlock()
 	klog.V(4).Infof("NodePublishVolume called with req: %#v", req)
 
-	// Validate Arguments
+	// Validate the request object
+	if err := validateNodePublishVolumeRequest(req); err != nil {
+		return nil, err
+	}
+
+	// Path to where we want to mount the volume inside the pod
```

Comment:
Can we assign these variable if and when they are needed in the function? There are multiple return cases before we get to the mounting stage. Just a good practice to avoid allocation memory for stuff that may not be needed

Reply:
Added the changes